src/Ormolu/Fixity/Internal.hs

```diff
@@ -1,12 +1,19 @@
 {-# LANGUAGE DeriveAnyClass #-}
 {-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternSynonyms #-}
 {-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ViewPatterns #-}
 
 module Ormolu.Fixity.Internal
-  ( FixityDirection (..),
+  ( OpName (..),
```

Comment:
Should the constructor be exported if we attach an invariant to the data inside? Perhaps we should only allow construction via smart constructors?

Reply:
Thanks for catching that, the pattern synonym exists exactly such that `OpName` can be opaque :smile: Will remove the constructor export.